public/app/features/trails/interactions.ts

```diff
@@ -0,0 +1,45 @@
+import { reportInteraction } from '@grafana/runtime';
+
+const PREFIX = 'grafana_explore_metrics_';
+
+type ExploreMetricsInteractions =
+  // User clicked on the "add to filters" button to add the label=value to the filters variable from a label breakdown panel
+  | 'breakdown_panel_label_filter_added'
+  // User clicked on the "select" button to select a label from a label breakdown panel
+  | 'breakdown_panel_label_selected'
+  // User changed the breakdown layout
+  // {layout: string}
+  | 'breakdown_layout_changed'
+  // User clicked on link the overview page to look at the breakdown of a label
+  | 'label_breakdown_link_clicked'
+  // User opened metric explore from dashboard
+  | 'embedded_drawer_opened'
+  // User clicks on the button to start a new exploration
+  | 'exploration_started'
+  // User clicks on an existing trail of bookmark from the homepage
+  | 'exploration_reloaded'
+  // User clicks on the delete bookmark button on the home page
+  | 'bookmark_deleted'
+  // User changes metric explore settings
+  // {stickyMainGraph: boolean}
+  | 'settings_changed'
+  // User clicks on history nodes to navigate exploration history
+  // Do we care about which trail they were on? Each should have a unique id. From step, to step?
+  | 'history_step_clicked'
+  // User clicks on tab to change the action view
+  // {view: string}
+  | 'metric_action_view_changed'
+  // User toggles the bookmark button on the current Explore Metric state
+  // {setTo: boolean}
+  | 'bookmark_toggled'
+  // User opens current seleted metric and filter query in "Explore"
+  | 'open_in_explore_clicked'
+  // User shares the URL for the current Explore Metric state
+  | 'share_url_clicked'
+  // User selects or unselects a metric
```

Comment:
do you mean both buttons will get tracked by the same event?
![Screenshot by Dropbox Capture](https://github.com/grafana/grafana/assets/143556099/867c7fdc-18a2-417e-9a2a-059f1d6ae0ea)

Reply:
Yes.

```
  // User selects or unselects a metric
  // {metric: metric | undefined}
  | 'metric_selected';
```

The `metric` will be `undefined` when you click "select new metric" and it will be an actual metric when the lower "Select" button is pressed.

Do we prefer a seperate event for the different buttons?

Reply:
I don't think we need a separate event. If we know `undefine` refers to `Select new metric` button, it is all good